src/core/p5.Renderer.js

```diff
@@ -49,6 +51,8 @@ class Renderer {
     this._clipping = false;
     this._clipInvert = false;
     this._curveTightness = 0;
+
+    this.currentShape = new Shape();
```

Comment:
@GregStanton it's not being used yet to actually draw shapes, but I've added a current shape to renderers. I've added a skeleton for a `.reset()` method that can be called on it when `beginShape()` is called to clear existing data, and I'm calling `updateShapeVertexProperties()` in a few places where the shape's data would need to be updated based on the renderer's state.